docs/development/architecture/PACKAGES.md

```diff
@@ -0,0 +1,77 @@
+---
+id: packages
+title: Packages
+sidebar_label: Packages
+---
+
+This page describes the top-level packages exported by the [typescript-eslint monorepo](https://github.com/typescript-eslint/typescript-eslint).
+Each of these are published as npm packages under the `@typescript-eslint` organization.
+
+## `@typescript-eslint/eslint-plugin`
+
+[`@typescript-eslint/eslint-plugin`] is the core [ESLint plugin](https://eslint.org/docs/user-guide/configuring/plugins) used by consumers to load in custom rules and rule configurations lists from typescript-eslint.
+Those rules rely on ESLint using the `@typescript-eslint/parser` package described below, and are generally built using the other packages on this page.
+
+## `@typescript-eslint/parser`
+
+[`@typescript-eslint/parser`] takes in ESLint configuration settings, reads in TypeScript source text, and produces an ESTree AST.
+This is necessary because TypeScript produces a different, incompatible AST format to the one that ESLint requires to work.
+
+For example, this is not valid JavaScript code because it contains the `: number` type annotation:
+
+```ts
+let x: number = 1;
+```
+
+ESLint's native Espree parser would raise an error attempting to parse it.
+
+Additionally, because TypeScript is developed separately from ESLint, ESTree, and Espree, its AST also represents nodes differently in many cases.
```

Comment:
might be worth mentioning here or in the ASTs page that different tools use different AST representations because they optimise for different things.

TS's AST is optimised for its usecase of parsing incomplete code and typechecking.
ESTree is unoptimised and intended for "general purpose" use-cases of traversing the AST.